base/src/commonMain/kotlin/utils/MeasurementUnit.kt

```diff
@@ -0,0 +1,320 @@
+        Grain -> value / GRAINS_IN_KILOGRAM
+        UsTon -> value / US_TONES_IN_KILOGRAM
+        ImperialTon -> value / UK_TONES_IN_KILOGRAM
+        CubicMeter -> TODO()
+        Liter -> value
+        Milliliter -> value / MILLILITERS_IN_LITER
+        UsLiquidGallon -> value / US_GALLONS_IN_LITER
+        UsLiquidQuart -> value / US_QUARTS_IN_LITER
+        UsLiquidPint -> value / US_PINTS_IN_LITER
+        UsLegalCup -> TODO()
+        UsFluidOunce -> value / US_FLUID_OUNCES_IN_LITER
+        UsFluidDram -> TODO()
+        ImperialGallon -> TODO()
+        ImperialQuart -> value / UK_QUARTS_IN_LITER
+        ImperialPint -> value / UK_PINTS_IN_LITER
+        ImperialCup -> TODO()
+        ImperialFluidOunce -> value / UK_FLUID_OUNCES_IN_LITER
+        ImperialFluidDram -> TODO()
+        CubicFoot -> TODO()
+        CubicInch -> TODO()
+    }
+}
+
+/**
+ * Translates from base SI to other systems
+ *
+ * @param value Value in base SI system
+ * @param toUnit The type of the return value
+ */
+private fun fromBaseSi(value: Decimal, toUnit: ScientificUnit<*, *>): Double {
+    return when (toUnit) {
+        Decimeter -> TODO()
+        Centimeter -> value * CENTIMETERS_IN_METER
+        Millimeter -> value * MILLIMETERS_IN_METER
+        Decameter -> TODO()
+        Hectometer -> TODO()
+        Kilometer -> value * KILOMETERS_IN_METER
+        Meter -> value
+        Inch -> value * INCHES_IN_METER
+        Foot -> value * FEET_IN_METER
+        Yard -> value * YARDS_IN_METER
+        Mile -> value * MILES_IN_METER
+        Celsius -> value - KELVIN_FREEZING
+        Kelvin -> value
+        Fahrenheit -> (value - KELVIN_FREEZING) * 9.0.toDecimal() / 5.0.toDecimal() + FAHRENHEIT_FREEZING
+        Tonne -> value * TONES_IN_KILOGRAM
+        Kilogram -> value
+        Gram -> value * GRAMS_IN_KILOGRAM
+        Milligram -> value * MILLIGRAMS_IN_KILOGRAM
+        Microgram -> TODO()
+        Stone -> value * STONES_IN_KILOGRAM
+        Pound -> value * POUNDS_IN_KILOGRAM
+        Ounce -> value * OUNCES_IN_KILOGRAM
+        Dram -> value * DRAMS_IN_KILOGRAM
+        Grain -> value * GRAINS_IN_KILOGRAM
+        UsTon -> value * US_TONES_IN_KILOGRAM
+        ImperialTon -> value * UK_TONES_IN_KILOGRAM
+        CubicMeter -> TODO()
+        Liter -> value
+        Milliliter -> value * MILLILITERS_IN_LITER
+        UsLiquidGallon -> value * US_GALLONS_IN_LITER
+        UsLiquidQuart -> value * US_QUARTS_IN_LITER
+        UsLiquidPint -> value * US_PINTS_IN_LITER
+        UsLegalCup -> TODO()
+        UsFluidOunce -> value * US_FLUID_OUNCES_IN_LITER
+        UsFluidDram -> TODO()
+        ImperialGallon -> value * UK_GALLONS_IN_LITER
+        ImperialQuart -> TODO()
+        ImperialPint -> TODO()
+        ImperialCup -> TODO()
+        ImperialFluidOunce -> value * UK_FLUID_OUNCES_IN_LITER
+        ImperialFluidDram -> TODO()
+        CubicFoot -> TODO()
+        CubicInch -> TODO()
+    }.toDouble()
+}
+
+private fun fromTo(
```

Comment:
private fun <Type : MeasurementType> ScientificUnit<*, Type>.convert(value: Double, to: ScientificUnit<*, Type>) should also work. No need to give implementations for Length/Weight etc specifically